examples/medplum-scheduling-demo/src/pages/AppointmentsPage.tsx

```diff
@@ -0,0 +1,96 @@
+import { Paper, Tabs } from '@mantine/core';
+import { Filter, Operator, SearchRequest } from '@medplum/core';
+import { MemoizedSearchControl } from '@medplum/react';
+import { useEffect, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+// Custom hook to manage the active tab and SearchRequest parameter for the
+// MemoizedSearchControl component
+function useTab(): [string, (tab: string | null) => void, SearchRequest, (definition: SearchRequest) => void] {
```

Comment:
I'm worried that the custom hook approach might be a bit advanced for our novice users to pick up. How much does the code change if this code is brought into the main component?